packages/cf-util-http/src/http.js

```diff
@@ -33,84 +25,136 @@ export function beforeSend(callback) {
   beforeSendCallbacks.push(callback);
 }
 
+export function clearBeforeSend() {
+  beforeSendCallbacks = [];
+}
+
+function toQueryParams(kvs) {
+  const queryParams = [];
+  // Clones the input
+  if (Array.isArray(kvs) || kvs instanceof Map) {
+    for (let [k, v] of kvs) {
+      queryParams.push([k, v]);
+    }
+  } else {
+    for (let k in kvs) {
+      if (kvs.hasOwnProperty(k)) {
+        queryParams.push([k, kvs[k]]);
+      }
+    }
+  }
+  return queryParams
+    .map(([k, v]) => `${encodeURIComponent(k)}=${encodeURIComponent(v)}`)
+    .join('&');
+}
+
+function toHash(headers) {
+  const hash = {};
+  for (let [k, v] of headers) {
+    hash[k] = v;
+    hash[k.toLowerCase()] = v;
+  }
+  return hash;
+}
+
+function wrapResponse(headers, status, body, text, response) {
+  return {
+    headers,
+    status,
+    body: headers['content-type'] === 'application/json'
+      ? JSON.parse(text)
```

Comment:
I would recommend `response.json().then()`
JSON.parse could throw, we don't handle this